### PR TITLE
GPU support and couple of addons

### PR DESCRIPTION
**Pull request by henryruhs**
1. GPU support added via `import torch` (don't ask me why)
2. Quit if model file is not present
3. Add processing time outputs
4. Fix CPU multi-thread newline bug

### Benchmark

Intel Core i9 13900k (Waterblock)

```
...................................................................................................................
Processing time: 34.57 seconds
```

### Runtime

```
Applied providers: ['CPUExecutionProvider'], with options: {'CPUExecutionProvider': {}}
inswapper-shape: [1, 3, 128, 128]
```

## GPU

GigaByte 4090 GAMING OC 24G (Waterblock)

### Benchmark

```
...................................................................................................................
Processing time: 9.70 seconds
```

### Runtime

```
Applied providers: ['CUDAExecutionProvider', 'CPUExecutionProvider'], with options: {'CPUExecutionProvider': {}, 'CUDAExecutionProvider': {'device_id': '0', 'gpu_mem_limit': '18446744073709551615', 'gpu_external_alloc': '0', 'gpu_external_free': '0', 'gpu_external_empty_cache': '0', 'cudnn_conv_algo_search': 'EXHAUSTIVE', 'cudnn_conv1d_pad_to_nc1d': '0', 'arena_extend_strategy': 'kNextPowerOfTwo', 'do_copy_in_default_stream': '1', 'enable_cuda_graph': '0', 'cudnn_conv_use_max_workspace': '1', 'tunable_op_enable': '0', 'enable_skip_layer_norm_strict_mode': '0', 'tunable_op_tuning_enable': '0'}}
inswapper-shape: [1, 3, 128, 128]
```

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,7 +1,8 @@
 numpy==1.24.3
 opencv-python==4.7.0.72
 onnx==1.14.0
-onnxruntime==1.15.0
```
Comment (s0md3v) [inline requirements.txt:4]: is there a reason behind removing onnxruntime?

People without a GPU will need it.

File: run.py [hunk 2]
```diff
@@ -31,7 +34,8 @@
 parser.add_argument('-t', '--target', help='replace this face', dest='target_path')
 parser.add_argument('-o', '--output', help='save output to this file', dest='output_file')
 parser.add_argument('--keep-fps', help='maintain original fps', dest='keep_fps', action='store_true', default=False)
-parser.add_argument('--gpu', help='use gpu', dest='gpu', action='store_true', default=False)
+if torch.cuda.is_available():
```
Comment (s0md3v) [inline run.py:37]: This isn't a good idea. I have an AMD GPU, I don't have CUDA - I have ROCM.

**Review by s0md3v — CHANGES_REQUESTED** [review]

**Comment by s0md3v** [comment]
Thanks for yet another amazing PR @henryruhs ❤️ 

Lets go through the few issues I marked and we are good to go!

**Comment by FurkanGozukara** [comment]
i think we can implement currently for nvidia with case checking and amd can be added later

File: requirements.txt (hunk 1 above)
Comment (henryruhs) [reply requirements.txt:4]: It does not work with both libs installed for some reason, took me hours to find out.

File: run.py (hunk 2 above)
Comment (henryruhs) [reply run.py:37]: From what I found in the internet this should still work for Radeon GPUs, can you give it a try?

**Comment by henryruhs** [comment]
Sorry but I call it a day, it's 2 o'clock at night here... you can still merge the PR and adjust the code later on.

**Review by s0md3v — APPROVED** [review]

**Pull request merged**